### PR TITLE
deps: upgrade to Fedora 38

### PR DESCRIPTION
**Pull request by malt3**
### Proposed change(s)

- mkosi: Use Fedora 38 as base
- mkosi: No longer downgrade GCP kernel
- Upgrade bazeldnf packages for linking against Fedora 38 libraries
- Upgrade containers to Fedora 38

### Additional info

As a reviewer, please test Constellation (create + initialize a real cluster) on a CSP of your choice. I did the majority of my tests on GCP so far.
You can use the following debug image for testing: `ref/deps-image-fedora-38/stream/debug/v2.9.0-pre.0.20230613084544-eeea7b1f56f4`

### Checklist

- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone
- [x] [Build Constellation binaries](https://github.com/edgelesssys/constellation/actions/runs/5253242711)
- [x] [Unit tests](https://github.com/edgelesssys/constellation/actions/runs/5253252813)
- [x] [OS image build](https://github.com/edgelesssys/constellation/actions/runs/5253871729)
- [x] [e2e AWS](https://github.com/edgelesssys/constellation/actions/runs/5257424928)
- [x] [e2e Azure](https://github.com/edgelesssys/constellation/actions/runs/5257441328)
- [x] [e2e GCP](https://github.com/edgelesssys/constellation/actions/runs/5257448355)

### CODE REVIEW

File: bazel/container/Containerfile [hunk 1]
```diff
@@ -8,8 +8,8 @@ ARG BAZELISK_VERSION=v1.16.0
 ARG BAZELISK_SHA256=168851e70cf5f95c0e215e7f3aaca5132ffc3c8dd8f585a4157b0be2b53cfe32
 
 ADD --checksum=sha256:${BAZELISK_SHA256} \
-    https://github.com/bazelbuild/bazelisk/releases/download/${BAZELISK_VERSION}/bazelisk-${TARGETOS}-${TARGETARCH} \
-    /usr/local/bin/bazelisk
+	https://github.com/bazelbuild/bazelisk/releases/download/${BAZELISK_VERSION}/bazelisk-${TARGETOS}-${TARGETARCH} \
+	/usr/local/bin/bazelisk
```
Comment (daniel-weisse) [inline bazel/container/Containerfile:12]: Intentional change from spaces to tabs?
This file seems to only use tabs for spacing now so its fine, though our other source files use spaces instead.

Comment (malt3) [reply bazel/container/Containerfile:12]: Done by my editor when i hit save. We should probably just adopt some form of auto formatting. Couldn't find a clear style guide anywhere for Dockerfiles yet.

**Review by elchead — APPROVED** [review]
tested on Azure and all is working

**Review by daniel-weisse — APPROVED** [review]
Worked with GCP

**Pull request merged**